docker_release/main.py

```diff
@@ -46,6 +52,18 @@ def _get_docker_image_name(docker_image_dir):
     return (path.basename(path.dirname(docker_image_dir)), path.basename(docker_image_dir))
 
 
+def _check_labelled(docker, image, repo):
+    sha = _get_current_sha(repo)
+    inspect = docker.inspect_image(image)
+    labels = inspect['Config']['Labels']
+    try:
+        label_hash = labels[GIT_HASH]
+        if label_hash != sha:
+            raise UserMessageException('Image %s hash mismatch in label %s. Expected %s, found %s' % (image, GIT_HASH, sha, label_hash))
+    except KeyError as e:
+        raise UserMessageException('Image %s missing %s label' % (image, e))
```

Comment:
how the current workflow looks like?

The aim of this script was to automate the common workflow of making a change of the image. Once the `Dockerfile` is changed (or any other image source file) then you could run `docker-release` to rebuild, tag and push the image. Now It looks like I need to tag the image somewhere outside of this script. So does it mean that image is built also outside of this script (with `docker build`? If so what is the reason of using `docker-relase` I would just use simply `docker push` instead.